src/routes/login.ts

```diff
@@ -65,6 +66,48 @@ export const createLoginRoute: RouteCreator =
     return frontend
       .getLoginFlow({ id: flow, cookie: req.header("cookie") })
       .then(({ data: flow }) => {
+        if (flow.ui.messages && flow.ui.messages.length > 0) {
+          // the login requires that the user verifies their email address before logging in
+          if (flow.ui.messages.some(({ id }) => id === 4000010)) {
+            // we will create a new verification flow and redirect the user to the verification page
+            frontend
+              .createBrowserVerificationFlow({
+                returnTo:
+                  (return_to && return_to.toString()) || flow.return_to || "",
+              })
+              .then(({ headers, data: verificationFlow }) => {
+                // we need the csrf cookie from the verification flow
+                res.setHeader("set-cookie", headers["set-cookie"])
+                // encode the verification flow id in the query parameters
+                const verificationParameters = new URLSearchParams({
+                  flow: verificationFlow.id,
+                  message: JSON.stringify(flow.ui.messages),
+                })
+                // redirect to the verification page with the custom message
+                res.redirect(
+                  "/verification?" + verificationParameters.toString(),
+                )
+              })
+              .catch(
+                redirectOnSoftError(
+                  res,
+                  next,
+                  getUrlForFlow(
+                    kratosBrowserUrl,
+                    "verification",
+                    new URLSearchParams({
+                      return_to:
+                        (return_to && return_to.toString()) ||
+                        flow.return_to ||
+                        "",
+                    }),
+                  ),
+                ),
+              )
+            return
+          }
+        }
+
```

Comment:
I find these `then().catch()` chains harder & harder to read the longer they get. Could you refactor this to an `async` method instead? I have been doing the same whenever I was working on these repos.